feature-cancel-insurance/src/main/kotlin/com/hedvig/android/feature/cancelinsurance/navigation/CancelInsuranceDestinations.kt

```diff
@@ -0,0 +1,68 @@
+package com.hedvig.android.feature.cancelinsurance.navigation
+
+import androidx.compose.material3.windowsizeclass.WindowSizeClass
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavHostController
+import androidx.navigation.navOptions
+import com.hedvig.android.feature.cancelinsurance.CancelInsuranceViewModel
+import com.hedvig.android.feature.cancelinsurance.InsuranceId
+import com.hedvig.android.feature.cancelinsurance.ui.result.CancellationSuccessDestination
+import com.hedvig.android.feature.cancelinsurance.ui.terminationdate.TerminationDateDestination
+import com.kiwi.navigationcompose.typed.Destination
+import com.kiwi.navigationcompose.typed.composable
+import com.kiwi.navigationcompose.typed.createRoutePattern
+import com.kiwi.navigationcompose.typed.navigate
+import com.kiwi.navigationcompose.typed.navigation
+import com.kiwi.navigationcompose.typed.popUpTo
+import kotlinx.serialization.Serializable
+import org.koin.androidx.compose.koinViewModel
+import org.koin.core.parameter.parametersOf
+
+internal sealed interface Destinations : Destination {
+  @Serializable
+  object CancelInsurance : Destinations
+}
+
+internal sealed interface CancelInsuranceDestinations : Destination {
+  @Serializable
+  object TerminationDate : CancelInsuranceDestinations
+
+  @Serializable
+  object TerminationSuccess : CancelInsuranceDestinations
+}
+
+internal fun NavGraphBuilder.cancelInsuranceGraph(
```

Comment:
This "graph" is what would be exposed from this module to the :app module, which would add all these paths to the graph of possible routes to navigate to.
I envision this future where each module can expose one graph like this, and the :app module can just add all of them to one NavHost, for it to have access to navigate to all of them.
For now though, the activity itself can simply host their own NavHost, which *only* has this one graph, of the destinations defined above.

Reply:
Thats great. I just ran into this as well when building the summary screens. Ill look at this when doing more work on that feature!

Reply:
Yeah I figured at this point as we do more stuff like this we might benefit from taking a few moments to think how to improve our experience with navigation. I think this time is as good as it gets, we're gonna touch navigation soon and the androidx navigation library is in a good spot to try this out with from what I've seen/read and I think I have a plan in my head that could work. Just need to try it out a bit and see how it feels.